features/steps/definitions.py

```diff
@@ -529,20 +529,46 @@ def create_a_jumphost(context, jumphost_name: str, network_name: str, keypair_na
         assert created_jumphost, f"Jumphost with name {jumphost_name} was not created successfully"
         context.collector.jumphosts.append(server.id)
 
-    ###### from SshSteps:
 
     @given("I have deployed a VM with IP {vm_ip_address}")
     def initialize(context, vm_ip_address: str):
         context.vm_ip_address = vm_ip_address
+    
+
+    @given("I have deployed {jh_quantity:d} JHs")
+    def initialize(context, jh_quantity):
+        context.jh=tools.collect_jhs(context.client,context.test_name, context.logger)
+        assert len(context.jh)>=jh_quantity, f"Not enough Jumphost with name found"
 
-    @given("I have a private key at {vm_private_ssh_key_path}")
-    def check_private_key_exists(context, vm_private_ssh_key_path: str):
+    
+    @given("I have a private key at {vm_private_ssh_key_path} for {username}")
+    def check_private_key_exists(context, vm_private_ssh_key_path: str, username:str):
         context.vm_private_ssh_key_path = vm_private_ssh_key_path
+        context.vm_username = username
         assert os.path.isfile(vm_private_ssh_key_path)
-
-    @then("I should be able to SSH into the VM as user {username}")
-    def test_ssh_connection(context, username):
-        ssh_client = SshClient(context.vm_ip_address, username, context.vm_private_ssh_key_path)
+    
+    @then ("I should be able to SSH into {jh_quantity:d} JHs and test their {conn_test} connectivity")
+    def step_iterate_steps(context, jh_quantity,conn_test:str):
+        context.assertline=None
+        for i in range(1, jh_quantity + 1):
+            if not isinstance(context.jh,str):
+                context.vm_ip_address=context.jh[i-1]['ip']            
```

Comment:
why are you using i-1? why do you want to start counting from 0, in that case change the range from (0, jh_quantity)

Reply:
thanks, didn't see that solution